src/vmm/src/arg_parser.rs

```diff
@@ -0,0 +1,129 @@
+use crate::arg_parser::CfgArgParseError::UnknownArg;
+use std::collections::HashMap;
+use std::fmt;
+use std::fmt::Formatter;
+use std::str::FromStr;
+
+#[derive(Debug, PartialEq)]
+pub(crate) enum CfgArgParseError {
```

Comment:
Wondering if `pub(crate)` is needed. This is very specific to `config` parsing so may be it should be something like `config/mod.rs`, `config/parser.rs` and then `pub(super)` or completely private might also work? 

The current `config.rs` will move to `config/mod.rs` . This way later all the individual configs may be moved to their own modules. eg. `config/memory.rs` etc.

Reply:
Good idea!